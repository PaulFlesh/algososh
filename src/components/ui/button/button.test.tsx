import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Button render", () => {
  it("Button render with text", () => {
      render(<Button text={"Some text"} />);
      expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("Button render without text", () => {
      render(<Button />);
      expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("Disabled button render", () => {
      render(<Button disabled={true} />);
      expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("Loading button render", () => {
      render(<Button isLoader={true} />);
      expect(screen.getByRole("button")).toMatchSnapshot();
  });

  it("Button render with callback", () => {
      const func = jest.fn();
      render(<Button onClick={func} />);
      fireEvent.click(screen.getByRole("button"));
      expect(func).toHaveBeenCalled();
  });
});
