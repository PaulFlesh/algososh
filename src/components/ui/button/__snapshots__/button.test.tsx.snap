// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render Button render with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    Some text
  </p>
</button>
`;

exports[`Button render Button render without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button render Disabled button render 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button render Loading button render 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;
